src/rust/engine/fs/store/src/remote.rs

```diff
@@ -93,6 +100,10 @@ impl LoadDestination for Vec<u8> {
   }
 }
 
+/// Blanket-implemented trait to approximate a `dyn AsyncRead + AsyncSeek + ...` trait object
+pub trait StoreSource_: AsyncRead + AsyncSeek + Send + Sync + Unpin + 'static {}
```

Comment:
Does this actually need to be generic, or would it be reasonable to hardcode in `tokio::fs::File` here? If we got rid of `store_bytes` it might be another story, but as long as we have the special facility for smaller blobs, the large blob facility will always be a file.

Reply:
Yeah, you're right. That's probably worth the simplification. Using a trait here makes the tests slightly simpler (can pass in a `std::io::Cursor` instead of having to create the temporary file), but that's not worth all these backflips.

Reply:
(Using the trait also allowed testing more error cases more easily... but if we specialise to `tokio::fs::File` those error cases may become unlikely/impossible, anyway.)